src/main/java/com/amplitude/Amplitude.java

```diff
@@ -108,11 +117,17 @@ public synchronized void flushEvents() {
       eventsToSend.clear();
       CompletableFuture.supplyAsync(
           () -> {
-            Response response = HttpCall.syncHttpCallWithEventsBuffer(eventsInTransit, apiKey);
+            Response response = null;
+            if (httpCallMode == Constants.httpCallMode.REGULAR_HTTPCALL) {
+              response =
+                  new GeneralHttpCall(eventsInTransit, apiKey).syncHttpCallWithEventsBuffer();
+            } else {
+              response = new BatchHttpCall(eventsInTransit, apiKey).syncHttpCallWithEventsBuffer();
+            }
             Status status = response.status;
 
             if (Retry.shouldRetryForStatus(status)) {
-              Retry.sendEventsWithRetry(eventsInTransit, apiKey, response);
+              Retry.sendEventsWithRetry(eventsInTransit, apiKey, response, httpCallMode);
```

Comment:
we don't need to send apiKey in this case, right? Since the new object already have it

Reply:
in Retry.retryEventsOnce, we have another HTTP call. So I think we still need to pass the apiKey?

Reply:
yeah, that's for you are passing the `httpCallMode`, instead of `HttpCall` right? And you need to construct the `httpCall` later, but if you don't need the events in `HttpCall`, do you still need to pass them to construct a new `HttpCall` in Retry?